src/watertap_contrib/reflo/unit_models/tests/test_solar_still.py

```diff
@@ -0,0 +1,247 @@
+#################################################################################
+# WaterTAP Copyright (c) 2020-2024, The Regents of the University of California,
+# through Lawrence Berkeley National Laboratory, Oak Ridge National Laboratory,
+# National Renewable Energy Laboratory, and National Energy Technology
+# Laboratory (subject to receipt of any required approvals from the U.S. Dept.
+# of Energy). All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/watertap-org/watertap/"
+#################################################################################
+
+import os
+import pytest
+
+from pyomo.environ import (
+    ConcreteModel,
+    value,
+    assert_optimal_termination,
+    units as pyunits,
+)
+from pyomo.network import Port
+from pyomo.util.check_units import assert_units_consistent
+
+from idaes.core import FlowsheetBlock, UnitModelCostingBlock, MaterialFlowBasis
+from idaes.core.util.testing import initialization_tester
+from idaes.core.util.exceptions import InitializationError, ConfigurationError
+from idaes.core.util.model_statistics import (
+    degrees_of_freedom,
+    number_variables,
+    number_total_constraints,
+    number_unused_variables,
+)
+from idaes.core.util.scaling import (
+    calculate_scaling_factors,
+    unscaled_variables_generator,
+)
+
+from watertap.core.solvers import get_solver
+from watertap.property_models.multicomp_aq_sol_prop_pack import (
+    MCASParameterBlock,
+)
+
+from watertap_contrib.reflo.unit_models import SolarStill
+from watertap_contrib.reflo.costing import TreatmentCosting
+
+# Get default solver for testing
+solver = get_solver()
+
+__location__ = os.path.realpath(os.path.join(os.getcwd(), os.path.dirname(__file__)))
+test_data_path = f"{__location__}/solar_still_test_data_phoenix_az.csv"
+
+
+def build_ss():
+
+    rho = 1000 * pyunits.kg / pyunits.m**3
+    inlet_dict = {
+        "solute_list": ["TDS"],
+        "mw_data": {"TDS": 31.4038218e-3},
+        "material_flow_basis": MaterialFlowBasis.mass,
+    }
+
+    water_yield_calc_dict = dict(
+        input_weather_file_path=test_data_path,  # path to input weather file
+        interval_day=15,  # interval at which to calculate water yield (e.g., every 15 days)
+        salinity=20,  # salinity of influent water; g/L
+        water_depth_basin=0.02,  # depth of water in solar still basin; m
+        length_basin=0.6,  # length of each side of basin (length=width); m
```

Comment:
I know this naming is an attempt to follow IDAES' typical naming conventions, but since these do not correspond to any establish IDAES names, I would opt to go with more readable names, such as `basin_water_depth` and `basin_length`.

Reply:
Another question is whether I need to specify these design variables here or if I can set Param/Var values how I typically would on an IDAES model.

Reply:
Pretty much same answer as before - we would be passing the initialized values in `build` to calculate water yield and if the user changes the value for each of these after the build, the calculated water yield would not reflect those different parameter values.

I suppose we could separate the two (i.e., remove the water yield calculation to a separate method that would need to be called manually after build):

```
m.fs.solar_still = SolarStill()
m.fs.solar_still.water_depth_basin.set_value(0.05)
m.fs.solar_still.water_depth_basin.length_basin(0.5)
m.fs.solar_still.calculate_water_yield()
```